test/quantization/test_numeric_suite_fx.py

```diff
@@ -429,3 +442,117 @@ def test_compare_model_outputs_lstm_dynamic_fx(self):
             lstm_input,
             lstm_hidden,
         )
+
+class TestFXGraphMatcher(QuantizationTestCase):
+    # TODO(future PR): more tests
+
+    def test_conv_mod_fp32_prepared_vs_int8(self):
+        m = nn.Sequential(nn.Conv2d(1, 1, 1)).eval()
+        mp = prepare_fx(m, {'': torch.quantization.default_qconfig})
+        # TODO(future PR): prevent the need for copying here, we can copy the
+        # modules but should reuse the underlying tensors
+        mp_copy = copy.deepcopy(mp)
+        mq = convert_fx(mp_copy)
+        results = get_matching_node_pairs(mp, mq)
+
+        expected_types = {'0': (nn.Conv2d, nnq.Conv2d)}
+        self.assert_types_for_matched_node_pairs(results, expected_types, mp, mq)
+
+    def test_linear_func_fp32_prepared_vs_int8(self):
+        class M(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.w = nn.Parameter(torch.Tensor(1, 1))
+                self.b = nn.Parameter(torch.Tensor(1))
+
+            def forward(self, x):
+                return F.linear(x, self.w, self.b)
+
+        m = M().eval()
+        mp = prepare_fx(m, {'': torch.quantization.default_qconfig})
+        # TODO(future PR): prevent the need for copying here, we can copy the
+        # modules but should reuse the underlying tensors
+        mp_copy = copy.deepcopy(mp)
+        mq = convert_fx(mp_copy)
+        results = get_matching_node_pairs(mp, mq)
+
+        expected_types = {'linear_1': (F.linear, toq.linear)}
+        self.assert_types_for_matched_node_pairs(results, expected_types, mp, mq)
+
+    def test_matching_failure_node_count(self):
```

Comment:
nit: can we spilt the test cases to:
- TestFXGraphMatcher: general functionality test
- TestFXGraphMatcherOps: test for individual or fused ops
- TestFXGraphMatcherModels: test for small or large models
Also I saw "fp32_prepared_vs_int8" appear multiple times, I think this is related to TestFXGraphMatcherOps/TestFXGraphMatcherModels maybe we can have separate test case for each pair of comparison as well

Reply:
sure, will do in this PR

Reply:
> Also I saw "fp32_prepared_vs_int8" appear multiple times, I think this is related to TestFXGraphMatcherOps/TestFXGraphMatcherModels maybe we can have separate test case for each pair of comparison as well

oh, by the way, "fp32_prepared" vs "int8" is the only pair of models which is expected to work with the current code.  The code should be generic to allow int8 vs fp32_prepared (switched order), as well as adding more model types.  That is not tested yet and saved for future PRs.